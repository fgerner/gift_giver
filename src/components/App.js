import React, {Component} from 'react';
import {button} from 'react-bootstrap';
import './App.css';


class App extends Component {
    constructor() {
        super();
        this.state = {gifts: []};
    }

    addGift = () => {
        const {gifts} = this.state;
        const ids = this.state.gifts.map(gift => gift.id);
        const maxId = ids.length > 0 ? Math.max(...ids) : 0;

        gifts.push({ id: maxId + 1});
        this.setState({gifts});
    }

    render() {
        return (
            <div className="App">
                <h2>Gift Giver</h2>
                <button className={'btn-add'} onClick={this.addGift} >Add Gift</button>
            </div>
        );
    }
}

export default App;
