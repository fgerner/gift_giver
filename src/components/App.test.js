import React from 'react';
import toJson from "enzyme-to-json";
import App from './App';
import {shallow} from 'enzyme';

const wrapper = shallow(<App />);

describe('App Component', () => {
    it('renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})
