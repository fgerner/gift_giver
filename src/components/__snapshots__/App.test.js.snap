// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="App"
>
  <h2>
    Gift Giver
  </h2>
  <div
    className="gift-list"
  />
  <button
    className="btn-add"
    onClick={[Function]}
  >
    Add Gift
  </button>
</div>
`;
